File: regexlib/python_re2_test_file/regexlib_5236.py
# 5236
# \{\\\*\\bkmkstart\s(.*?)\}
# POLYNOMIAL
# nums:5
# POLYNOMIAL AttackString:""+"{\\*\\bkmkstart "*5000+"! _1SLQ_2"

import re2 as re
from time import perf_counter

regex = """\{\\\*\\bkmkstart\s(.*?)\}"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "{\\*\\bkmkstart " * i * 10000 + "! _1SLQ_2"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")

File: regexlib/javascript_test_file/regexlib_5236.js
/* 5236
 * \{\\\*\\bkmkstart\s(.*?)\}
 * POLYNOMIAL
 * nums:3
 * POLYNOMIAL AttackString:""+"{\\*\\bkmkstart "*5000+"! _1SLQ_2"
 */
var REGEX = /\{\\\*\\bkmkstart\s(.*?)\}/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + '{\\*\\bkmkstart '.repeat(i*10000) + '! _1SLQ_2'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 10000 + ": " + time_cost+" ms")
}